spec/controllers/api/v1/conversations_controller_spec.rb

```diff
@@ -0,0 +1,142 @@
+        expect(JSON.parse(response.body, symbolize_names: true)[:data][:meta][:all_count]).to eq(1)
+      end
+    end
+  end
+
+  describe 'GET /api/v1/conversations/:id' do
+    let(:conversation) { create(:conversation, account: account) }
+
+    context 'when it is an unauthenticated user' do
+      it 'returns unauthorized' do
+        get "/api/v1/conversations/#{conversation.display_id}"
+
+        expect(response).to have_http_status(:unauthorized)
+      end
+    end
+
+    context 'when it is an authenticated user' do
+      let(:agent) { create(:user, account: account, role: :agent) }
+
+      it 'shows the conversation' do
+        get "/api/v1/conversations/#{conversation.display_id}",
+            headers: agent.create_new_auth_token,
+            as: :json
+
+        expect(response).to have_http_status(:success)
+        expect(JSON.parse(response.body, symbolize_names: true)[:meta][:contact_id]).to eq(conversation.contact_id)
+      end
+    end
+  end
+
+  describe 'POST /api/v1/conversations/:id/toggle_status' do
+    let(:conversation) { create(:conversation, account: account) }
+
+    context 'when it is an unauthenticated user' do
+      it 'returns unauthorized' do
+        post "/api/v1/conversations/#{conversation.display_id}/toggle_status"
+
+        expect(response).to have_http_status(:unauthorized)
+      end
+    end
+
+    context 'when it is an authenticated user' do
+      let(:agent) { create(:user, account: account, role: :agent) }
+
+      it 'toggles the conversation status' do
+        expect(conversation.status).to eq('open')
+
+        post "/api/v1/conversations/#{conversation.display_id}/toggle_status",
+             headers: agent.create_new_auth_token,
+             as: :json
+
+        expect(response).to have_http_status(:success)
+        expect(conversation.reload.status).to eq('resolved')
+      end
+    end
+  end
+
+  describe 'POST /api/v1/conversations/:id/update_last_seen' do
+    let(:conversation) { create(:conversation, account: account) }
+
+    context 'when it is an unauthenticated user' do
+      it 'returns unauthorized' do
+        post "/api/v1/conversations/#{conversation.display_id}/update_last_seen"
+
+        expect(response).to have_http_status(:unauthorized)
+      end
+    end
+
+    context 'when it is an authenticated user' do
+      let(:agent) { create(:user, account: account, role: :agent) }
+
+      it 'updates last seen' do
+        params = { agent_last_seen_at: '-1' }
+
+        post "/api/v1/conversations/#{conversation.display_id}/update_last_seen",
+             headers: agent.create_new_auth_token,
+             params: params,
+             as: :json
+
+        expect(response).to have_http_status(:success)
+        expect(conversation.reload.agent_last_seen_at).to eq(DateTime.strptime(params[:agent_last_seen_at].to_s, '%s'))
+      end
+    end
+  end
+
+  describe 'GET /api/v1/conversations/:id/get_messages' do
```

Comment:
Thanks for bringing this API to notice, we are not using this API, could you please remove this endpoint.

Reply:
@pranavrajs done :)